src/main/java/org/opensearch/security/securityconf/impl/v6/ConfigV6.java

```diff
@@ -356,4 +357,32 @@ public String toString() {
 
     }
 
+    public static class OnBehalfOfSettings {
```

Comment:
I will answer the question 1by1:
1. For the naming yes, I forgot to change the name in V7 :/, so good catch, and I have fixed it.
2. Since the v6 we are not using it, so I didn't update it. But it is making sense that both of them should have the same structure. So I just made the update in V6 to match the V7.